crates/wasmtime/src/module.rs

```diff
@@ -406,7 +405,11 @@ impl Module {
         let (mmap, info) =
             wasmtime_jit::finish_compile(translation, obj, funcs, trampolines, tunables)?;
 
-        Ok((mmap, Some(info)))
+        Ok((
+            mmap,
+            Some(info),
+            is_branch_protection_enabled(&engine.compiler().isa_flags()),
```

Comment:
Instead of ferrying around this `bool` as a return value from this function could the `is_branch_protection_enabled` testing go as a method on `Engine`? Most contexts that are being given this `bool` already have the `Engine` so they should be able to query the engine itself for this